added app/v1api20250101

GeraldLoeffler opened the pull request:
## What this PR does

Added support for [Microsoft.App ARM REST API version 2025-01-01](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/app/resource-manager/Microsoft.App/stable/2025-01-01).

## How does this PR make you feel?

uneasy - first PR against ASO and the code generator is a big mystery to me.

## Checklist

- [x] this PR contains documentation
- [x] this PR contains YAML Samples

theunrepentantgeek commented:
> the code generator is a big mystery to me.

It's definitely a complex beast, no two ways about it, and our documentation is more focused on the operator itself. If you have any specific questions, let me know - I'm happy to go into detail, and will update our docs as well.

GeraldLoeffler commented:
Note: it's unexpected that the integration tests do not make use of the YAML samples, which I also had to provide. It seems the integration tests just build the same resource manifests as in my YAML samples, just using Go instead of YAML.

GeraldLoeffler commented:
I've updated the PR as far as I can: added integration tests. But I can't record their execution, because the corporate Azure environment I have access to doesn't allow that. I'm submitting as is - what's missing is `task controller:test-integration-envtest`.

GeraldLoeffler commented:
@microsoft-github-policy-service agree

theunrepentantgeek commented:
Thanks for the contribution @GeraldLoeffler, much appreciated.
